tools/wiki_entities/README.md

```diff
@@ -12,8 +12,27 @@ pip install mafan
 pip install textblob
 ```
 
-4. Run the following two scripts. These two scripts contain some simple rules designed for English, Spanish, and Chinese. You may want to modify them to fit the new language **X**. You will find "**X**/wiki_quality.txt" and "**X**/wiki_all.txt".
+4. Run the script. This script contain some simple rules designed for English, Spanish, and Chinese. You may want to modify them to fit the new language **X**. You will find "**X**/wiki_quality.txt" and "**X**/wiki_all.txt".
 ```
-python wiki_all_phrase_filter.py -lang X
-python high_quality_phrase_filter.py -lang X
+python extract_entities.py --lang X --quality
+python extract_entities.py --lang X --all
+```
+
+4.1. If neither `--quality` nor `--all` intend to be used, the script also supports custom arguments:
+```
+python extract_entities.py -h
+
+usage: extract_entities.py [-h] [--lang LANG] [--min_sup MIN_SUP] [--min_perc MIN_PERC] [--stopwords_orig] [--stopwords_end] [--only_alpha] [--quality] [--all]
+
+optional arguments:
+  -h, --help           show this help message and exit
+  --lang LANG          language/dir containing files
+  --min_sup MIN_SUP    minimum support for considering entity
+  --min_perc MIN_PERC  minimum percentage for considering entity
+  --stopwords_orig     whether tokens starting w/ stopwords or w/ high proportion of stopwords should be ignored
+  --stopwords_end      whether tokens ending w/ stopwords should be ignored
+  --only_alpha         whether only alphanumeric (ignoring only numeric) tokens should be extracted
+  --quality            whether quality profile should be used
+  --all                whether all profile should be used
```

Comment:
it is not natural to have complex dependency among flags. better to use a flag like --mode=quality/all/custom to indicate the mutual exclusiveness.
Also, please add "custom_" as prefix to those flags that depend on custom mode

Reply:
Ok.
Even if I create a flag like `--mode=quality/all/custom`, args `--all` and `--stopwords_orig` would still have mutual exclusiveness. In this case, is it ok for me to ignore `--stopwords_orig`? Or would it be better to create a new script `extract_entities_custom.py` allowing these fine-grained flags?

Reply:
why don't we make stopwords_orig and stopwords_end not mode dependent?
btw, shall we rename `stopwords_orig` as `skip_head_token_stopword` and `stopwords_end` as `skip_end_token_stopword`

Reply:
> btw, shall we rename stopwords_orig as skip_head_token_stopword and stopwords_end as skip_end_token_stopword

Indeed better names.

> why don't we make stopwords_orig and stopwords_end not mode dependent?

I don't agree much to this because these modes were trying to imitate what the old scripts did (`quality_...py` and `wiki_all_....py`). And the old scripts had `skip_head_token_stopword` as dependency. If these flags are mode independent, there would be mutual exclusion.

I keep my suggestion of maintaining one script for `all/quality` and other script allowing the user choose whatever predicate it suits or `MIN_SUP`, `MIN_PERC` (but without the modes options).

Reply:
Thanks. This lgtm.